test/integration/targets/vmware_vmotion/tasks/main.yml

```diff
@@ -0,0 +1,79 @@
+# Test code for the vmware_vmotion module.
+# Copyright: (c) 2019, Pavan Bidkar <pbidkar@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- import_role:
+    name: prepare_vmware_tests
+  vars:
+    setup_attach_host: true
+    setup_virtualmachines: True
+    setup_datastore: true
+
+- name: Create VMs
+  vmware_guest:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ dc1 }}"
+    validate_certs: no
+    name: test_vm1
+    folder: vm
+    state: present
+    guest_id: centos7_64Guest
+    disk:
+    - size_gb: 1
+      type: thin
+      datastore: '{{ rw_datastore }}'
+    hardware:
+      version: latest
+      memory_mb: 1024
+      num_cpus: 1
+      scsi: paravirtual
+  register: vm_create
+
+- name: Perform vMotion of virtual machine
+  vmware_vmotion:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    validate_certs: no
+    vm_name: test_vm1
+    destination_host: '{{ esxi1 }}'
```

Comment:
Here, the VM is created on esxi1, and so the vMotion is ignored (changed==False).

```
TASK [vmware_vmotion : Create VMs] ***********************************************************************************************************************************************************
task path: /home/goneri/git_repos/ansible_projects/a/test/results/.tmp/integration/vmware_vmotion-g6vdrekn-ÅÑŚÌβŁÈ/test/integration/targets/vmware_vmotion/tasks/main.yml:12
<testhost> ESTABLISH LOCAL CONNECTION FOR USER: goneri
<testhost> EXEC /bin/sh -c 'echo ~goneri && sleep 0'
<testhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/goneri/.ansible/tmp/ansible-tmp-1574115544.09139-28502037121925 `" && echo ansible-tmp-1574115544.09139-28502037121925="` echo /home/goneri/.ansible/tmp/ansible-tmp-1574115544.09139-28502037121925 `" ) && sleep 0'                                                                                                      
Using module file /home/goneri/git_repos/ansible_projects/a/lib/ansible/modules/cloud/vmware/vmware_guest.py
<testhost> PUT /home/goneri/.ansible/tmp/ansible-local-198890nbt4ssz/tmpjvhljags TO /home/goneri/.ansible/tmp/ansible-tmp-1574115544.09139-28502037121925/AnsiballZ_vmware_guest.py
<testhost> EXEC /bin/sh -c 'chmod u+x /home/goneri/.ansible/tmp/ansible-tmp-1574115544.09139-28502037121925/ /home/goneri/.ansible/tmp/ansible-tmp-1574115544.09139-28502037121925/AnsiballZ_vmware_guest.py && sleep 0'                                                                                                                                                                    
<testhost> EXEC /bin/sh -c 'ESXI1_HOSTNAME=esxi1.test ESXI1_PASSWORD='"'"'!234AaAa56'"'"' ESXI1_USERNAME=root ESXI2_HOSTNAME=test2.test ESXI2_PASSWORD='"'"'!234AaAa56'"'"' ESXI2_USERNAME=root VCENTER_HOSTNAME=vcenter.test VCENTER_PASSWORD='"'"'!234AaAa56'"'"' VCENTER_USERNAME=administrator@vsphere.local VMWARE_VALIDATE_CERTS=false /tmp/python-hcjd6127-ansible/python /home/goneri/.ansible/tmp/ansible-tmp-1574115544.09139-28502037121925/AnsiballZ_vmware_guest.py && sleep 0'                                                                                              
<testhost> EXEC /bin/sh -c 'rm -f -r /home/goneri/.ansible/tmp/ansible-tmp-1574115544.09139-28502037121925/ > /dev/null 2>&1 && sleep 0'
changed: [testhost] => {
    "changed": true,
    "instance": {
        "annotation": "",
        "current_snapshot": null,
        "customvalues": {},
        "guest_consolidation_needed": false,
        "guest_question": null,
        "guest_tools_status": "guestToolsNotRunning",
        "guest_tools_version": "0",
        "hw_cluster": "DC0_C0",
        "hw_cores_per_socket": 1,
        "hw_datastores": [
            "LocalDS_0"
        ],
        "hw_esxi_host": "esxi1.test",
        "hw_files": [
            "[LocalDS_0] test_vm1/test_vm1.vmx",
            "[LocalDS_0] test_vm1/test_vm1.vmsd",
            "[LocalDS_0] test_vm1/test_vm1.vmdk"
        ],
        "hw_folder": "/DC0/vm",
        "hw_guest_full_name": null,
        "hw_guest_ha_state": null,
        "hw_guest_id": null,
        "hw_interfaces": [],
        "hw_is_template": false,
        "hw_memtotal_mb": 1024,
        "hw_name": "test_vm1",
        "hw_power_status": "poweredOff",
        "hw_processor_count": 1,
        "hw_product_uuid": "421ba001-b24c-888a-4131-d1a33419b54e",
        "hw_version": "vmx-14",
        "instance_uuid": "501bf7d0-174d-54d3-c0e9-291fe5a300a7",
        "ipv4": null,
        "ipv6": null,
        "module_hw": true,
        "moid": "vm-41",
        "snapshots": [],
        "vimref": "vim.VirtualMachine:vm-41",
        "vnc": {}
    },
    "invocation": {
        "module_args": {
            "annotation": null,
            "cdrom": [],
            "cluster": null,
            "convert": null,
            "customization": {},
            "customization_spec": null,
            "customvalues": [],
            "datacenter": "DC0",
            "datastore": null,
            "delete_from_inventory": false,
            "disk": [
                {
                    "datastore": "LocalDS_0",
                    "size_gb": 1,
                    "type": "thin"
                }
            ],
            "esxi_hostname": null,
            "folder": "vm",
            "force": false,
            "guest_id": "centos7_64Guest",
            "hardware": {
                "memory_mb": 1024,
                "num_cpus": 1,
                "scsi": "paravirtual",
                "version": "latest"
            },
            "hostname": "vcenter.test",
            "is_template": false,
            "linked_clone": false,
            "name": "test_vm1",
            "name_match": "first",
            "networks": [],
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 443,
            "proxy_host": null,
            "proxy_port": null,
            "resource_pool": null,
            "snapshot_src": null,
            "state": "present",
            "state_change_timeout": 0,
            "template": null,
            "use_instance_uuid": false,
            "username": "administrator@vsphere.local",
            "uuid": null,
            "validate_certs": false,
            "vapp_properties": [],
            "wait_for_customization": false,
            "wait_for_ip_address": false,
            "wait_for_ip_address_timeout": 300
        }
    }
}

TASK [vmware_vmotion : Perform vMotion of virtual machine] ***********************************************************************************************************************************
task path: /home/goneri/git_repos/ansible_projects/a/test/results/.tmp/integration/vmware_vmotion-g6vdrekn-ÅÑŚÌβŁÈ/test/integration/targets/vmware_vmotion/tasks/main.yml:34
<testhost> ESTABLISH LOCAL CONNECTION FOR USER: goneri
<testhost> EXEC /bin/sh -c 'echo ~goneri && sleep 0'
<testhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/goneri/.ansible/tmp/ansible-tmp-1574115547.2926908-2814510963515 `" && echo ansible-tmp-1574115547.2926908-2814510963515="` echo /home/goneri/.ansible/tmp/ansible-tmp-1574115547.2926908-2814510963515 `" ) && sleep 0'                                                                                                   
Using module file /home/goneri/git_repos/ansible_projects/a/lib/ansible/modules/cloud/vmware/vmware_vmotion.py
<testhost> PUT /home/goneri/.ansible/tmp/ansible-local-198890nbt4ssz/tmpokb7kzje TO /home/goneri/.ansible/tmp/ansible-tmp-1574115547.2926908-2814510963515/AnsiballZ_vmware_vmotion.py
<testhost> EXEC /bin/sh -c 'chmod u+x /home/goneri/.ansible/tmp/ansible-tmp-1574115547.2926908-2814510963515/ /home/goneri/.ansible/tmp/ansible-tmp-1574115547.2926908-2814510963515/AnsiballZ_vmware_vmotion.py && sleep 0'                                                                                                                                                                
<testhost> EXEC /bin/sh -c 'ESXI1_HOSTNAME=esxi1.test ESXI1_PASSWORD='"'"'!234AaAa56'"'"' ESXI1_USERNAME=root ESXI2_HOSTNAME=test2.test ESXI2_PASSWORD='"'"'!234AaAa56'"'"' ESXI2_USERNAME=root VCENTER_HOSTNAME=vcenter.test VCENTER_PASSWORD='"'"'!234AaAa56'"'"' VCENTER_USERNAME=administrator@vsphere.local VMWARE_VALIDATE_CERTS=false /tmp/python-hcjd6127-ansible/python /home/goneri/.ansible/tmp/ansible-tmp-1574115547.2926908-2814510963515/AnsiballZ_vmware_vmotion.py && sleep 0'                                                                                           
<testhost> EXEC /bin/sh -c 'rm -f -r /home/goneri/.ansible/tmp/ansible-tmp-1574115547.2926908-2814510963515/ > /dev/null 2>&1 && sleep 0'
ok: [testhost] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "destination_datastore": null,
            "destination_host": "esxi1.test",
            "destination_resourcepool": null,
            "hostname": "vcenter.test",
            "moid": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 443,
            "proxy_host": null,
            "proxy_port": null,
            "use_instance_uuid": false,
            "username": "administrator@vsphere.local",
            "validate_certs": false,
            "vm_name": "test_vm1",
            "vm_uuid": null
        }
    },
    "running_host": "esxi1.test"
}

TASK [vmware_vmotion : assert that changes were made] ****************************************************************************************************************************************
task path: /home/goneri/git_repos/ansible_projects/a/test/results/.tmp/integration/vmware_vmotion-g6vdrekn-ÅÑŚÌβŁÈ/test/integration/targets/vmware_vmotion/tasks/main.yml:44
fatal: [testhost]: FAILED! => {
    "assertion": "vm_vmotion is changed",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```

It sounds like you should either pin the VM deployment on esxi2.

Reply:
Hi @goneri , Thanks for catching this. I suspected this. Changed VM names in other 2tests as well to test cross cluster VMotion
Thanks